integration-tests/src/test/java/com/datastax/oss/driver/api/core/metadata/SerializableMetadataIT.java

```diff
@@ -0,0 +1,105 @@
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.Optional;
+import org.apache.commons.lang3.SerializationUtils;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+
+@Category(ParallelizableTests.class)
+public class SerializableMetadataIT {
+
+  private static final CcmRule ccmRule = CcmRule.getInstance();
+
+  private static final SessionRule<CqlSession> sessionRule = SessionRule.builder(ccmRule).build();
+
+  @ClassRule public static final TestRule chain = RuleChain.outerRule(ccmRule).around(sessionRule);
+
+  @Before
+  public void setupKeyspace() {
+    List<String> statements =
+        ImmutableList.of(
+            "CREATE KEYSPACE IF NOT EXISTS numbers WITH REPLICATION = { 'class' : 'NetworkTopologyStrategy', 'datacenter1' : 1 }",
+            "CREATE TABLE numbers.all (val int, doub int, str text, primary key (val,doub)) WITH CLUSTERING ORDER BY (doub DESC)",
+            "CREATE TYPE numbers.numbers_data (doub int, str text)",
+            "CREATE TABLE numbers.smaller (val int primary key, data numbers_data)",
+            "CREATE INDEX doub_idx on numbers.all (doub)",
+            "CREATE OR REPLACE FUNCTION numbers.avgState ( state tuple<int,bigint>, val int ) CALLED ON NULL INPUT RETURNS tuple<int,bigint> LANGUAGE java AS"
+                + " 'if (val !=null) { state.setInt(0, state.getInt(0)+1); state.setLong(1, state.getLong(1)+val.intValue()); } return state;'",
+            "CREATE OR REPLACE FUNCTION numbers.avgFinal ( state tuple<int,bigint> ) CALLED ON NULL INPUT RETURNS double LANGUAGE java AS"
+                + " 'double r = 0; if (state.getInt(0) == 0) return null; r = state.getLong(1); r/= state.getInt(0); return Double.valueOf(r);'",
+            "CREATE AGGREGATE IF NOT EXISTS numbers.average ( int ) SFUNC avgState STYPE tuple<int,bigint> FINALFUNC avgFinal INITCOND (0,0)",
+            "INSERT INTO numbers.all (val,doub,str) VALUES (1,2,'one')",
+            "INSERT INTO numbers.all (val,doub,str) VALUES (2,4,'two')",
+            "INSERT INTO numbers.all (val,doub,str) VALUES (3,6,'three')",
+            "CREATE MATERIALIZED VIEW IF NOT EXISTS numbers.big AS SELECT FROM numbers.all"
+                + " WHERE val is not null and doub is not null and val > 2"
+                + " PRIMARY KEY (val,doub)");
+
+    statements.stream().forEach((stmt) -> sessionRule.session().execute(stmt));
+  }
+
+  @Test
+  public void should_contain_serializable_metadata() {
+
+    Optional<KeyspaceMetadata> ksOption =
+        sessionRule.session().getMetadata().getKeyspace("numbers");
+    assertTrue(ksOption.isPresent());
+    KeyspaceMetadata ks = ksOption.get();
+    assertTrue(ks instanceof DefaultKeyspaceMetadata);
+
+    /* Validate that the keyspace metadata is fully populated */
```

Comment:
"fully populated" here means "all the maps in DefaultKeyspaceMetadata have data in them".  This definition is based on the impl of the class and therefore isn't super-flexible.  I'd love to have a definition based on the class (i.e. if the concrete instance had some kind of isFullyPopulated() method or something) but doing so seemed to lead down some unpleasant avenues... so I decided to leave it at this for now.

I'm certainly open to better suggestions here.

Reply:
Would that notion of "fully populated" be of any use outside of tests though?

If this is more about how deep we should go with the checks, see my suggestion in another comment to merge this with `DescribeIT`.

Reply:
No, the idea of "fully populated" doesn't really have any meaning outside of tests.  That's part of the reason I'm not lobbying for it... it feels like a @VisibleForTesting kind of thing of limited utility

I'll look into this more as I look to migrating this stuff to DescribeIT